### ISSUE TITLE
Updating Conan requirement

### ISSUE THREAD

**Issue by ajjackson** [brille/brille#100]
We are getting test failures for Euphonic on Mac, because the Mac testers use a newer version of GCC that is not supported by the pinned version of Conan.
https://github.com/pace-neutrons/Euphonic/actions/runs/9777689258/job/26992904951?pr=303

A short-term fix would be to update the Conan requirement to 1.64 instead of the current 1.57 pin.

However, an incompatible Conan 2.x API is now being used so in the long run deeper changes might be needed.

**Comment by g5t** [comment]
I think moving to Conan 2 is the right thing to do; #101 does just that but needs a bit more work.
In the mean time, are you able to check if the MacOS wheel works on the Euphonic test machine?

I don't think you have access to workflow artifacts, so I've attached them here.
[cp39-macosx.zip](https://github.com/user-attachments/files/16094955/cp39-macosx.zip)
[cp38-macosx.zip](https://github.com/user-attachments/files/16094957/cp38-macosx.zip)
[cp312-macosx.zip](https://github.com/user-attachments/files/16094958/cp312-macosx.zip)
[cp311-macosx.zip](https://github.com/user-attachments/files/16094959/cp311-macosx.zip)
[cp310-macosx.zip](https://github.com/user-attachments/files/16094960/cp310-macosx.zip)

**Comment by ajjackson** [comment]
I'll see if I can smuggle those into a test workflow! Euphonic testing is all on Github Actions now.

Thanks for looking into this so quickly: I tried building against an older GCC in the test Action but this seems to be causing more problems than it solves 😭.

**Comment by g5t** [comment]
I probably should have followed the first link you shared before. Euphonic is using `macos-latest` runners where we're stuck on `macos-12` here -- and the two differ not only in operating system version but also processor architecture.

The runner was _building_ brille, so I guess it's unlikely that the wheels I shared will work. I'm arranging interactive access to an M1 Mac to investigate the status of using pre-built wheels.

Can you modify the action or Euphonic dependency (in a test-branch?) to pull in the `100-conan` branch of this repository?

**Comment by ajjackson** [comment]
It is installing correctly with 100-conan, but the numerical results are different on Mac 😞 
https://github.com/pace-neutrons/Euphonic/actions/runs/9810005711/job/27089171105?pr=304

**Comment by g5t** [comment]
I'll try to take a look at it but fear I might get lost. 

The first thing that pops out at me is that you're building Euphonic's binary module with LLVM clang obtained from brew.
I'm trying to use cibuildwheel to make all of the wheels for brille, and encountering warnings about mixing compilers and/or Python 3.8 binary sources on the `macos-latest` runners.

I recall Becking working out the need to install `llvm-openmp/11.1.0` when building macOS wheels for brille due to incompatibilities between later versions and other packages (I think numpy).

Have you gone with the whole LLVM 18.1.8 suite for a similar reason? Maybe related to numpy 2.0?

In any case, I doubt that the version of brille built from source in that CI workflow uses the brew-sourced clang; so that could be the problem.

I should have cibuildwheel produced wheels in short order. Maybe using one of those instead of building from source will fix the problem?

**Comment by ajjackson** [comment]
I think you're right, Brille uses the system clang while Euphonic looks for homebrew. It is possible to override the Euphonic compiler choice by setting CC/CXX (and marking these as variables to pass through tox environment...) but when I tried using this to build with an old GCC
- the euphonic build got grumpy (I think there is further work to do if you want to use an old compiler environment from homebrew, and maybe euphonic doesn't like GCC on Mac?)
- it wasn't entirely clear that Brille picked up the same compiler anyway

Ok, while we figure this out I will go ahead and switch Euphonic to test/build with macos-12 as well, so that this doesn't hold up a Numpy 2 compatibility release.

**Comment by g5t** [comment]
Getting the actions workflows and CMake/Python builds in shape on the current cibuildwheel and runner architectures has been more involved than I anticipated -- but is nearing completion. 

Two things:

1. During the process, I found that a test of interpolation on the C++ side of things gave different results for macos-14 than any other platform. I'm not entirely certain where in the depths of the standard library or arithmetic implementation the difference comes from, but it seems to be the result of rounding differences and a too-fragile test case.
Namely, the test interpolates a vector field that does not have all of the symmetries of the crystal and attempts to avoid that fact by only picking points in the fully invariant portion of the Brillouin zone; unfortunately I picked points on the surface of the irreducible polyhedron and rounding somewhere moved them 'outside' such that one of the point group operations is applied to the interpolated result.
This should not be a problem when it interpolating any function that has all of the symmetries of the crystal, otherwise brille could never work.
_I haven't looked in detail, but could this 'surface' effect explain the test failure in Euphonic?_ The eigenvectors get jumbled by the symmetry operations and brille _should_ correct their permutation; but maybe it doesn't do so correctly (while hopefully not modifying the resulting intensity, or other observables)?
2. The process would be finished, except that I haven't enforced that the wheels are numpy >= 2 compatible; and in testing I was forced at one point to downgrade numpy to install a test brille wheel.
If your goal is numpy 2 compatibility _and_ you still want to use brille; I'll need to do some more work before pushing wheels to PyPI.

**Comment by ajjackson** [comment]
My understanding is that packages built against Numpy 2 ABI should be _also_ (binary) compatible with 1.26

**Comment by g5t** [comment]
brille [`v0.8.0`](https://pypi.org/project/brille/) wheels are available for both Intel and Arm macOS, so the Euphonic tests shouldn't need to build the binary module for `macos-12`, `macos-13`, or `macos-14` runners with Python 3.8 to 3.12.

Does the test still fail with the new wheels?

**Comment by ajjackson** [comment]
It looks like the relevant tests are now passing on both `macos-12` and `macos-14`. The CI is still failing because some new warnings appeared; these may be from the latest version of spglib. But I think on the Brille front it all looks good 👍 Thanks for sorting this out so quickly.

**Comment by ajjackson** [comment]
Yup, all green over here https://github.com/pace-neutrons/Euphonic/pull/304
